pkg/loki/modules.go

```diff
@@ -1160,7 +1167,7 @@ func (t *Loki) initUsageReport() (services.Service, error) {
 }
 
 func (t *Loki) deleteRequestsClient(clientType string, limits *validation.Overrides) (deletion.DeleteRequestsClient, error) {
-	if !t.supportIndexDeleteRequest() {
+	if !t.supportIndexDeleteRequest() || !t.Cfg.CompactorConfig.RetentionEnabled {
```

Comment:
again, this seems to add to the argument this additional check should be moved into that function.

Reply:
Same as above, it is more explicit for the reader that we need an index store that supports deletes and explicitly opt-in to use the feature.